### PR TITLE
PMM-4474 add rds exporter agent

### PR DESCRIPTION
**Pull request by percona-csalguero**

### CODE REVIEW

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-448/

**Comment by JNKPercona** [comment]
http://docs-test.cd.percona.com/PR-448/

File: api/inventorypb/agents.proto [hunk 1]
```diff
@@ -38,6 +38,14 @@ enum AgentType {
   QAN_MYSQL_SLOWLOG_AGENT = 8;
   QAN_MONGODB_PROFILER_AGENT = 9;
   QAN_POSTGRESQL_PGSTATEMENTS_AGENT = 10;
+  RDS_EXPORTER = 11;
+}
+
+// RDSInstanceType describbes supported DB types.
+enum RDSInstanceType {
```
Comment (AlekSi) [inline api/inventorypb/agents.proto]: Is it used somewhere?

File: api/inventorypb/agents.proto [hunk 2]
```diff
@@ -302,6 +310,34 @@ message QANPostgreSQLPgStatementsAgent {
...
+  // The pmm-agent identifier which runs this instance.
+  string pmm_agent_id = 2;
+  // Desired Agent status: enabled (false) or disabled (true).
+  bool disabled = 3;
+  // Service identifier.
+  string service_id = 4;
+  // Custom user-assigned labels.
+  map<string, string> custom_labels = 5;
+  // Node ID. We use it to get the node and from the node we can extract the AWS region
+  string node_id = 6;
```
Comment (AlekSi) [inline api/inventorypb/agents.proto]: We know Node ID from Service ID. We don't need a separate field.

File: api/inventorypb/agents.proto [hunk 3]
```diff
@@ -464,6 +502,56 @@ message ChangeMySQLdExporterResponse {
...
+    }
+  ];
+  // RDS username for scraping metrics.
+  string username = 3 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+  // RDS password for scraping metrics.
+  string password = 4;
```
Comment (AlekSi) [inline api/inventorypb/agents.proto]: Access and secret keys, not username and password.

File: api/inventorypb/agents.proto [hunk 4]
```diff
@@ -464,6 +502,56 @@ message ChangeMySQLdExporterResponse {
...
+  // Use TLS for database connections.
+  bool tls = 5;
+  // Skip TLS certificate and hostname validation.
+  bool tls_skip_verify = 6;
```
Comment (AlekSi) [inline api/inventorypb/agents.proto]: We don't actually connect to the database in rds_exporter. We don't need those fields.

File: api/inventorypb/agents.proto [hunk 5]
```diff
@@ -983,4 +1071,18 @@ service Agents {
       body: "*"
     };
   }
+  // AddRDSExporter adds rds_exporter Agent.
```
Comment (AlekSi) [inline api/inventorypb/agents.proto]: Please move it up, before `RemoveAgent`.

File: api/inventorypb/nodes.proto [hunk 6]
```diff
@@ -237,6 +256,48 @@ message RemoveNodeRequest {
...
+  map<string, string> custom_labels = 6;
+}
+
+message AddRemoteRDSNodeResponse {
+  RemoteNode remote = 1;
+}
+
+// Remove
+
+message RemoveRemoteRDSNodeRequest {
```
Comment (AlekSi) [inline api/inventorypb/nodes.proto]: We don't have type-specific Remove requests.

File: api/inventorypb/agents.proto (hunk 1 above)
Comment (percona-csalguero) [reply api/inventorypb/agents.proto]: Not yet. I'll remove it

**Comment by percona-csalguero** [comment]
Outdated branch. I'll close this one in favor of PMM-4474_Add_rds_exporter_agent-B

**Pull request closed without merging**